src/server/network.ts

```diff
@@ -172,6 +179,9 @@ export class Request {
 
   _updateWithRawHeaders(headers: types.HeadersArray) {
     this._headers = headers;
+    this._headersMap.clear();
+    for (const { name, value } of this._headers)
+      this._headersMap.set(name.toLowerCase(), value);
```

Comment:
Aren't we clobbering multiple `set-cookie` headers here?

Reply:
Well I guess that would be in the response, but still.

Reply:
Our array is not a real array, it comes from a map. Should be fixed when we fix our headers to be array...

Reply:
Failing test? :D

Reply:
I think thats my bug. I actually have a cookie parser for you somewhere. I'm looking for it.